app.js

```diff
@@ -0,0 +1,105 @@
+const session = require("express-session");
+// Initialize an express application
+const express = require('express');
+
+// Import the express-async-errors module to handle async errors
+require("express-async-errors");
+
+const app = express();
+// Set the view engine to ejs for rendering views
+app.set("view engine", "ejs");
+
+// Use body-parser middleware to parse incoming request bodies
+app.use(require("body-parser").urlencoded({ extended: true }));
+
+require("dotenv").config(); // to load the .env file into the process.env object
```

Comment:
Requires change:
Move this dotenv import to the top of the file. You'll want to ensure this loads first because some 3rd packages may expect a process.env to be defined before being imported.

Reply:
wow, I didn't think about this. Thank you for pointing this out